src/power_grid_model/_utils.py

```diff
@@ -439,11 +482,45 @@ def validate_data_filter(
 
 def is_sparse(component_data: ComponentData) -> bool:
     """Check if component_data is sparse or dense. Only batch data can be sparse."""
-    return isinstance(component_data, dict) and set(component_data.keys()) == {"indptr", "data"}
+    return isinstance(component_data, dict) and set(component_data.keys()) == {
+        "indptr",
+        "data",
+    }
 
 
 def is_columnar(component_data: ComponentData) -> bool:
     """Check if component_data is columnar or row based"""
     if is_sparse(component_data):
         return not isinstance(component_data["data"], np.ndarray)
     return not isinstance(component_data, np.ndarray)
+
+
+def validate_component_data(component_data: ComponentData, component=None) -> None:
+    """Checks if component_data is of ComponentData and raises ValueError if its not"""
+    component_name = f"'{component}'" if component is not None else ""
+    err_msg = f"Invalid data for {component_name} component. " "{0}"
+    err_msg_suffixed = err_msg + "It should be a 1D/2D Numpy structured array or a dictionary of such."
+
+    if is_sparse(component_data):
+        indptr = component_data["indptr"]
+        if not isinstance(indptr, np.ndarray):
+            raise TypeError(err_msg_suffixed.format(f"Invalid indptr type {type(indptr).__name__}. "))
+        sub_data = component_data["data"]
```

Comment:
idem to my comment on the `utils` about making a function that does this for you

Reply:
Added functions which shall make it nicer. Shall be used in the next PR.

Reply:
See https://github.com/PowerGridModel/power-grid-model/issues/749